OneCommunityGlobal/HighestGoodNetworkApp
# ShreyaV Unit Testing for Add Task Modal WBS Component

# Description
This PR addresses the testing and functionality coverage for the AddTaskModal component. It fixes multiple edge case scenarios, enhances the component’s accessibility, and ensures all features are thoroughly validated through unit tests.

## Related PRS (if any):
This frontend PR is independent but validates functionality critical to backend workflows. No associated backend PR is required.
…

## Main changes explained:
- Refactored AddTaskModal.jsx: Improved accessibility, added missing labels and identifiers, and ensured consistent state handling.
- Created test file: Located at WBS/WBSDetail/_tests_/AddTaskModal.test.jsx, it now includes comprehensive test cases for all key functionalities.
- Mocked TinyMCE Editor: Simplified testing of the Editor component by using a mock, enabling validation of editor-related features.
- Validation logic: Enhanced input validations and warning displays for hours and date pickers.
…

## How to test:
1. Check out the branch.
2. Run npm install to ensure dependencies are installed.
3. Clear site data/cache to avoid stale state issues.
4. Navigate to the frontend repository.
5. Run the following command to ensure all tests pass:
`npm test -- AddTaskModal.test.js`

## Screenshots or videos of changes:
<img width="1156" alt="Screenshot 2024-12-22 at 12 47 52 AM" src="https://github.com/user-attachments/assets/675ab950-50ad-468d-bd73-f14d5c85b3b9" />
